### PR TITLE
Fix/14856 network settings page

### LINKED ISSUE #14856: [Bug]: Network list is hidden and scrollable in Settings > Networks with small browser width
### Describe the bug

The network list in Settings > Networks has a small height using a small browser width. This results in a hidden list that requires the user to scroll to access. This issue does not apply to the popup view, only the browser tab view. 

Bug: 
https://user-images.githubusercontent.com/20778143/171966727-8f5a8beb-885e-4992-8dc5-d9b1b14db247.mov

Looks good in popup: 

### Steps to reproduce

1. Open MM in browser tab
2. Go to Settings > Networks
3. Minimize the width of the browser
4. Observe the network list UI

### Error messages or log output

_No response_

### Version

10.14.7

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Currently, the network page is broken on smaller browser sizes. This PR updates some styling to fix that as well as some small network indicator color improvements

## More Information

* Fixes #14856

### Before
MM-bug-settings-network-ui

Network indicator using multiple different grey colors 

before

### After

https://user-images.githubusercontent.com/8112138/173706078-ccd6892b-96e5-401e-bb55-013a5907f122.mov

Popover/Action view remains the same

https://user-images.githubusercontent.com/8112138/173894277-ad96276c-d214-495a-9ee4-fa449fe4d905.mov

Making network indicator more consistent 

Screen Shot 2022-06-14 at 4 25 56 PM

## Manual Testing Steps

- In expanded view
- Go to Settings > Nework
- Adjust to small browser size
- Also test in popover view

## Pre-Merge Checklist

- [x] PR template is filled out
- ~[x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added~ NA
- [x] PR is linked to the appropriate GitHub issue
- ~[x] PR has been added to the appropriate release Milestone~ NA

### + If there are functional changes:

- ~[x] Manual testing complete & passed~ NA
- ~[x] "Extension QA Board" label has been applied~ NA

### CODE REVIEW

**Comment by seaona** [comment]
I've QAd the fix and works great @georgewrmarshall

**Comment by darkwing** [comment]
QA loves it @georgewrmarshall ; just need a rebase!

File: ui/helpers/constants/design-system.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ export const COLORS = {
   TEXT_ALTERNATIVE: 'text-alternative',
   TEXT_MUTED: 'text-muted',
   ICON_DEFAULT: 'icon-default',
+  ICON_ALTERNATIVE: 'text-alternative',
   ICON_MUTED: 'icon-muted',
   BORDER_DEFAULT: 'border-default',
   BORDER_MUTED: 'border-muted',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.js:17]: Adding missing color to design system conts

File: ui/components/app/dropdowns/network-dropdown.js [hunk 2]
```diff
@@ -174,7 +174,7 @@ class NetworkDropdown extends Component {
             <div className="network-check__transparent">✓</div>
           )}
           <ColorIndicator
-            color={opts.isLocalHost ? 'localhost' : COLORS.TEXT_MUTED}
+            color={opts.isLocalHost ? 'localhost' : COLORS.ICON_MUTED}
             size={SIZES.LG}
             type={ColorIndicator.TYPES.FILLED}
           />
```
Comment (georgewrmarshall) [inline ui/components/app/dropdowns/network-dropdown.js:180]: The network indicators for custom networks uses multiple grays this is an attempt to consolidate them

## Before
before

## After
Screen Shot 2022-06-14 at 4 25 56 PM

File: ui/components/app/network-display/network-display.js [hunk 3]
```diff
@@ -48,9 +48,7 @@ export default function NetworkDisplay({
         >
           <ColorIndicator
             color={
-              networkType === NETWORK_TYPE_RPC
-                ? COLORS.ICON_DEFAULT
-                : networkType
+              networkType === NETWORK_TYPE_RPC ? COLORS.ICON_MUTED : networkType
             }
             size={indicatorSize}
             type={ColorIndicator.TYPES.FILLED}
```
Comment (georgewrmarshall) [inline ui/components/app/network-display/network-display.js:54]: Consolidating network indicators greys

File: ui/components/ui/color-indicator/color-indicator.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { COLORS, SIZES } from '../../../helpers/constants/design-system';
 export default function ColorIndicator({
   size = SIZES.SM,
   type = 'outlined',
-  color = COLORS.ICON_DEFAULT,
+  color = COLORS.ICON_MUTED,
   borderColor,
   iconClassName,
 }) {
```
Comment (georgewrmarshall) [inline ui/components/ui/color-indicator/color-indicator.js:12]: Consolidating network indicators greys. Same as above

File: ui/components/ui/text-field/index.scss [hunk 5]
```diff
@@ -2,3 +2,9 @@
 .MuiInputBase-input {
   color: var(--color-text-default) !important;
 }
+
+.MuiFormHelperText-root.Mui-error {
+  @include H7;
+
+  color: var(--color-error-default) !important;
+}
```
Comment (georgewrmarshall) [inline ui/components/ui/text-field/index.scss:10]: Fixes non design system approved color for input error text

Screen Shot 2022-06-14 at 4 39 20 PM

File: ui/pages/settings/networks-tab/index.scss [hunk 6]
```diff
@@ -291,34 +189,43 @@
...
   }
 
   &__add-network-form-footer {
     display: flex;
     flex-flow: row;
-    padding: 0.75rem 0;
+    padding: 12px 0;
     width: 60%;
 
     .btn-secondary {
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/index.scss:231]: The main guts of this PR fixes a lot of arbitrary media queries and widths in favour of using CSS grid and our breakpoint sass variables

File: ui/pages/settings/networks-tab/networks-form/networks-form.js [hunk 7]
```diff
@@ -551,6 +551,7 @@ const NetworksForm = ({
           iconFillColor="var(--color-warning-default)"
           useIcon
           withRightButton
+          className="networks-tab__add-network-form__alert"
         />
       ) : null}
       <div
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-form/networks-form.js:557]: Adding class to remove unnecessary margin

## Before
Screen Shot 2022-06-14 at 4 42 29 PM

## After

Screen Shot 2022-06-14 at 4 42 53 PM

File: ui/pages/settings/networks-tab/networks-list/networks-list.js [hunk 8]
```diff
@@ -66,7 +66,7 @@ const NetworksList = ({
       {searchQuery === '' && (
         <Typography
           variant={TYPOGRAPHY.H6}
-          margin={[6, 0, 0, 9]}
+          marginTop={4}
           color={COLORS.TEXT_ALTERNATIVE}
           className="networks-tab__networks-list__label"
         >
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-list/networks-list.js:72]: removing unnecessary margins

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 9]
```diff
@@ -73,9 +73,15 @@ const NetworksListItem = ({
...
+        />
       ) : (
-        <div className="networks-tab__content__check-icon__transparent">✓</div>
+        <IconCheck
+          className="networks-tab__content__icon-check"
+          color="transparent"
+        />
       )}
       {network.chainId in CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP ? (
         <Identicon
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:89]: Improving accessibility of the check icon as well as the visual representation of the"invisible" icon

File: ui/pages/settings/networks-tab/index.scss [hunk 10]
```diff
@@ -7,31 +7,34 @@
...
-    display: flex;
-    height: 100%;
-    max-width: 779px;
-    justify-content: space-between;
+    &--with-networks-list-popup-footer {
+      // padding-bottom required to offset content from fixed footer: &__networks-list-popup-footer
+      padding-bottom: 80px;
+      // isolation contains stacking context(z-index) of &__networks-list-popup-footer
+      isolation: isolate;
+    }
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/index.scss:15]: This is a conditional class that adds styles necessary to keep the content visible when a fixed footer is added and isolated the stacking context concerns to this div so the fixed footer will never overlap with any elements outside of `__content`

Screen Shot 2022-06-15 at 8 40 58 AM
Screen Shot 2022-06-15 at 8 41 58 AM

File: ui/pages/settings/networks-tab/networks-tab.js [hunk 11]
```diff
@@ -92,7 +93,12 @@ const NetworksTab = ({ addNewNetwork }) => {
       {isFullScreen ? (
         <NetworksFormSubheader addNewNetwork={addNewNetwork} />
       ) : null}
-      <div className="networks-tab__content">
+      <div
+        className={classnames('networks-tab__content', {
+          'networks-tab__content--with-networks-list-popup-footer':
+            !isFullScreen && !shouldRenderNetworkForm,
+        })}
+      >
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-tab.js:101]: Adds a conditional class that adds styles necessary to keep the content visible when a fixed footer is added and isolated the stacking context concerns to this div so the fixed footer will never overlap with any elements outside of `__content`

Screen Shot 2022-06-15 at 8 40 58 AM
Screen Shot 2022-06-15 at 8 41 58 AM

**Review by jpuri — APPROVED** [review]
Good work

**Review by darkwing — APPROVED** [review]
Tested it out and it's excellent!  Very nicely done!

**Pull request merged**